drivers/rtc/rtc_ds1337.c

```diff
@@ -0,0 +1,747 @@
+			pending = 1;
+		}
+	} else {
+		if (status_reg & DS1337_A2F_MASK) {
+			status_reg &= ~DS1337_A2F_MASK;
+			pending = 1;
+		}
+	}
+
+	err = i2c_reg_write_byte_dt(&config->i2c, DS1337_STATUS_REG, status_reg);
+	if (err) {
+		goto out_unlock;
+	}
+
+out_unlock:
+	ds1337_unlock_sem(dev);
+
+	if (err) {
+		return err;
+	}
+	return pending;
+}
+
+#ifdef DS1337_INT_GPIOS_IN_USE
+
+static int ds1337_alarm_set_callback(const struct device *dev, uint16_t id,
+				     rtc_alarm_callback callback, void *user_data)
+{
+	const struct ds1337_config *config = dev->config;
+	struct ds1337_data *data = dev->data;
+	uint8_t reg_val = 0;
+	uint8_t mask = 0;
+	int err;
+
+	if (config->gpio_int.port == NULL) {
+		return -ENOTSUP;
+	}
+
+	if (id >= DS1337_ALARMS_COUNT) {
+		LOG_ERR("Invalid alarm ID: %d", id);
+		return -EINVAL;
+	}
+
+	ds1337_lock_sem(dev);
+
+	data->alarm_callbacks[id] = callback;
+	data->alarm_user_data[id] = user_data;
+
+	/* Enable alarm interrupt if callback provided */
+	if (id == DS1337_ALARM_1_ID) {
+		mask = DS1337_A1IE_MASK;
+		reg_val = (callback != NULL) ? DS1337_A1IE_MASK : 0;
+	} else {
+		mask = DS1337_A2IE_MASK;
+		reg_val = (callback != NULL) ? DS1337_A2IE_MASK : 0;
+	}
+	err = i2c_reg_update_byte_dt(&config->i2c, DS1337_CONTROL_REG, mask, reg_val);
+
+	ds1337_unlock_sem(dev);
+
+	/* Alarm IRQ might have already been triggered */
+	(void)k_work_submit(&data->work);
+
+	return err;
+}
+
+#endif
+
+#endif
+
+static int ds1337_init(const struct device *dev)
+{
+	const struct ds1337_config *config = dev->config;
+	struct ds1337_data *data = dev->data;
+	uint8_t reg_val;
+	int err;
+
+	(void)k_sem_init(&data->lock, 1, 1);
+
+	if (!i2c_is_ready_dt(&config->i2c)) {
+		LOG_ERR("I2C bus not ready");
+		return -ENODEV;
+	}
+
+#ifdef DS1337_INT_GPIOS_IN_USE
+	if (config->gpio_int.port != NULL) {
+		if (!gpio_is_ready_dt(&config->gpio_int)) {
+			LOG_ERR("GPIO not ready");
+			return -ENODEV;
+		}
+
+		err = gpio_pin_configure_dt(&config->gpio_int, GPIO_INPUT);
+		if (err) {
+			LOG_ERR("Failed to configure interrupt GPIO, error: %d", err);
+			return err;
+		}
+
+		err = gpio_pin_interrupt_configure_dt(&config->gpio_int, GPIO_INT_EDGE_TO_ACTIVE);
+		if (err) {
+			LOG_ERR("Failed to enable GPIO interrupt, error: %d", err);
+			return err;
+		}
+
+		gpio_init_callback(&data->irq_callback, ds1337_irq_handler,
+				   BIT(config->gpio_int.pin));
+
+		err = gpio_add_callback_dt(&config->gpio_int, &data->irq_callback);
+		if (err) {
+			LOG_ERR("Failed to add GPIO callback, error: %d", err);
+			return err;
+		}
+
+		data->dev = dev;
+		data->work.handler = ds1337_work_callback;
+	}
+#endif
+
+	/* Display warning if alarm flags are set */
+	err = i2c_reg_read_byte_dt(&config->i2c, DS1337_STATUS_REG, &reg_val);
+	if (err) {
+		return err;
+	}
+	if (reg_val & DS1337_A1F_MASK) {
+		LOG_WRN("Alarm 0 might have been missed!");
+	}
+	if (reg_val & DS1337_A2F_MASK) {
+		LOG_WRN("Alarm 1 might have been missed!");
+	}
```

Comment:
```suggestion
	if (reg_val & DS1337_A1F_MASK) {
		LOG_WRN("Alarm 1 might have been missed!");
	}
	if (reg_val & DS1337_A2F_MASK) {
		LOG_WRN("Alarm 2 might have been missed!");
	}
```

Reply:
I was not sure whether I should refer to the alarms by RTC numbering or by IDs passed to alarm API functions, corrected as per your suggestion